omc3/check_corrections.py

```diff
@@ -0,0 +1,612 @@
+                         type=PathOrStr,
+                         help=("Path to the directory where to write the output files. "
+                              "If the input ``corrections`` input consists of multiple folders, their name will "
+                               "be used to sort the output data into subfolders.", ))
+    params.add_parameter(name="corrections",
+                         required=True,
+                         nargs="+",
+                         type=PathOrStr,
+                         help="Paths to the correction files/directories to use. "
+                              "If files are given, these will all be applied as corrections at the same time. "
+                              "If folders are given, these are assumed to individually containing "
+                              "the correction files. See then also ``file_pattern``.",)
+    params.add_parameter(name="file_pattern",
+                         help="Filepattern to use to find correction files in folders (as regex).",
+                         type=str,
+                         default=r"^changeparameters*?\.madx$",
+                         )
+    # Parameters (similar/same as in global correction) ---
+    params.add_parameter(name="optics_params",
+                         type=str,
+                         nargs="+",
+                         choices=OPTICS_PARAMS_CHOICES,
+                         help=f"List of parameters for which the cuts should be applied (e.g. {BETA}X {BETA}Y)", )
+    params.add_parameter(name="modelcut",
+                         nargs="+",
+                         type=float,
+                         help="Reject BPMs whose deviation to the model is higher "
+                              "than the corresponding input. Input in order of optics_params.",)
+    params.add_parameter(name="errorcut",
+                         nargs="+",
+                         type=float,
+                         help="Reject BPMs whose error bar is higher than the corresponding "
+                              "input. Input in order of optics_params.",)
+    params.add_parameter(name="beta_filename",
+                         default=CORRECTION_DEFAULTS["beta_filename"],
+                         help="Prefix of the beta file to use. E.g.: ``beta_phase_``", )
+    # Plotting -----------------------------------------------------------------
+    params.add_parameter(name="plot",
+                         action="store_true",
+                         help="Activate plotting."
+                         )
+    params.update(get_plotting_style_parameters())
+    return params
+
+
+@entrypoint(get_params())
+def correction_test_entrypoint(opt: DotDict, accel_opt) -> None:
+    """ Entrypoint function to test the given corrections.
+
+    .. todo:: Instead of writing everything out, it could return
+             dictionaries of the TFSDataFrames and Figures.
+             But I don't see a usecase at the moment (jdilly 2023)
```

Comment:
Could this be an optional parameter then? Defaults to False, but should someone want to use the function in another script they can.

Reply:
No, no additional parameter. Just return it always. Doesn't cost anything, but I was to lazy to implement it. Can be done by whoever needs it then